build/efr32mg12/efr32-app.mk

```diff
@@ -0,0 +1,524 @@
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+# 
+#   @file
+#         Common makefile definitions for building applications based
+#         on the Silicon Labs EFR32 SDK.
+#
+
+#
+#   This makefile is primarily intended to support building the
+#   OpenWeave example applications on Silicon Labs platforms.  However
+#   external developers should feel free to use it if they find
+#   it useful.
+#
+#   To build an application using this makefile, include the file
+#   in a project-specific Makefile, define make variables describing
+#   the application and how it should be built, and then call the
+#   GenerateBuildRules function.  E.g.:
+#   
+#       PROJECT_ROOT = $(realpath .)
+#
+#       BUILD_SUPPORT_DIR = $(PROJECT_ROOT)/third_party/openweave-core/build/efr32
+#       
+#       include $(BUILD_SUPPORT_DIR)/efr32-app.mk
+#       include $(BUILD_SUPPORT_DIR)/efr32-openweave.mk
+#       include $(BUILD_SUPPORT_DIR)/efr32-openthread.mk
+#       include $(BUILD_SUPPORT_DIR)/efr32-freertos.mk
+#       
+#       APP := openweave-efr32-bringup
+#       
+#       SRCS = \
+#           $(PROJECT_ROOT)/main.cpp \
+#           ...
+#        
+#       INC_DIRS = \
+#           $(PROJECT_ROOT) \
+#           ...
+#       
+#       LIBS = \
+#           ...
+#       
+#       DEFINES = \
+#           ...
+#       
+#       $(call GenerateBuildRules)
+#
+
+
+# ==================================================
+# Sanity Checks
+# ==================================================
+
+ifndef BOARD
+$(error ENVIRONMENT ERROR: BOARD not set)
+endif
+ifndef EFR32_SDK_ROOT
+$(error ENVIRONMENT ERROR: EFR32_SDK_ROOT not set)
+endif
+ifndef EFR32_TOOLS_ROOT
+$(error ENVIRONMENT ERROR: EFR32_TOOLS_ROOT not set)
+endif
+ifndef FREERTOS_ROOT
+$(error ENVIRONMENT ERROR: FREERTOS_ROOT not set)
+endif
+ifndef OPENTHREAD_ROOT
+$(error ENVIRONMENT ERROR: OPENTHREAD_ROOT not set)
+endif
+
+
+
+# Differentiate between boards
+# - BRD4304A / SLWSTK6000B / MGM12P Module / 2.4GHz@19dBm
+# - BRD4161A / SLWSTK6000B / Wireless Starter Kit / 2.4GHz@19dBm
+# - BRD4166A / SLTB004A / Thunderboard Sense 2 / 2.4GHz@10dBm
+# - BRD4170A / SLWSTK6000B / Multiband Wireless Starter Kit / 2.4GHz@19dBm, 915MHz@19dBm, 868MHz@19dBm
+ifeq ($(BOARD),BRD4304A)
+MCU = EFR32MG12P432F1024GM48
+else
+ifeq ($(BOARD),BRD4161A)
+MCU = EFR32MG12P432F1024GL125
+else
+ifeq ($(BOARD),BRD4166A)
+MCU = EFR32MG12P332F1024GL125
+else
+ifeq ($(BOARD),BRD4170A)
+MCU = EFR32MG12P433F1024GM68
+else
+$(error Please provide a value for BOARD variable e.g BOARD=BRD4161A (currently supported BRD4304A, BRD4161A, BRD4166A or BRD4170A))
```

Comment:
It is a good idea. This may require changes to OpenThread and how it uses the BOARD variable to select the hal config files.  As mentioned in comment further down we will be looking to restructure the EFR32 OpenThread platform code as there is a lot common between the different families.  As part of this we could look to add custom board support.  Currently you would have to alter the existing hal config files in OpenThread.

Reply:
Thanks for the explanation, Marvin.  We can leave this as-is for the current PR.